src/screens/SettingsScreen.js

```diff
@@ -0,0 +1,306 @@
+import React, { useState, useEffect } from 'react';
+import { Text, View, Button, StyleSheet, TextInput, Image } from 'react-native';
+import { supabase } from '../utils/hooks/supabase';
+import { useAuthentication } from '../utils/hooks/useAuthentication';
+
+export default function SettingsScreen() {
+  const { user } = useAuthentication();
+  const [loading, setLoading] = useState(true);
+  const [displayName, setDisplayName] = useState('');
+  const [email, setEmail] = useState('');
+  const [dateOfBirth, setDateOfBirth] = useState('01/01/1998'); // Default random date
+  const [profilePictureUrl, setProfilePictureUrl] = useState('');
+  const [initialDisplayName, setInitialDisplayName] = useState('');
+  const [initialEmail, setInitialEmail] = useState('');
+  const [initialProfilePictureUrl, setInitialProfilePictureUrl] = useState('');
+  const [editingDisplayName, setEditingDisplayName] = useState(false);
+  const [editingEmail, setEditingEmail] = useState(false);
+  const [editingDateOfBirth, setEditingDateOfBirth] = useState(false);
+  const [editingProfilePicture, setEditingProfilePicture] = useState(false);
+
+  useEffect(() => {
+    if (user !== null) {
+      fetchUserData();
+    }
+  }, [user]);
+
+  const fetchUserData = async () => {
+    try {
+      const { data, error } = await supabase
+        .from('profiles') // Replace with your table name
+        .select('*')
+        .eq('id', user.id)
+        .single();
+
+      if (error) throw error;
+
+      setDisplayName(data.username || user.email.split('@')[0]);
+      setEmail(data.email || user.email);
+      setDateOfBirth(data.dob || '01/01/1998');
+      setProfilePictureUrl(data.profile_picture_url || 'https://example.com/default-profile-picture.jpg'); // Default URL
+      setInitialDisplayName(data.username || user.email.split('@')[0]);
+      setInitialEmail(data.email || user.email);
+      setInitialProfilePictureUrl(data.profile_picture_url || 'https://example.com/default-profile-picture.jpg');
```

Comment:
Is this redundant? Looks like it calls a different function but does the same thing as line 40

Reply:
So we didn't work on the login page which didn't include a username section when you log in. So What we decided after talking to some other groups because they did have username that we'd use the first part of the email as a display name in the meantime.